azurerm/resource_arm_notification_hub_namespace.go

```diff
@@ -0,0 +1,245 @@
+		return fmt.Errorf("Error setting `sku`: %+v", err)
+	}
+
+	if props := resp.NamespaceProperties; props != nil {
+		d.Set("enabled", props.Enabled)
+		d.Set("namespace_type", props.NamespaceType)
+		d.Set("servicebus_endpoint", props.ServiceBusEndpoint)
+	}
+
+	return nil
+}
+
+func resourceArmNotificationHubNamespaceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).notificationNamespacesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["namespaces"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error deleting Notification Hub Namespace %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	// the future returned from the Delete method is broken 50% of the time - let's poll ourselves for now
+	// Related Bug: https://github.com/Azure/azure-sdk-for-go/issues/2254
+	log.Printf("[DEBUG] Waiting for Notification Hub Namespace %q (Resource Group %q) to be deleted", name, resourceGroup)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"200", "202"},
+		Target:  []string{"404"},
+		Refresh: notificationHubNamespaceStateRefreshFunc(ctx, client, resourceGroup, name),
+		Timeout: 10 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Notification Hub %q (Resource Group %q) to be deleted: %s", name, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for the deletion of Notification Hub Namespace %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func expandNotificationHubNamespacesSku(input []interface{}) *notificationhubs.Sku {
+	v := input[0].(map[string]interface{})
+
+	skuName := v["name"].(string)
+
+	return &notificationhubs.Sku{
+		Name: notificationhubs.SkuName(skuName),
+	}
+}
+
+func flattenNotificationHubNamespacesSku(input *notificationhubs.Sku) []interface{} {
+	outputs := make([]interface{}, 0)
+	if input == nil {
+		return outputs
+	}
+
+	output := map[string]interface{}{
+		"name": string(input.Name),
+	}
+	outputs = append(outputs, output)
+	return outputs
+}
+
+func notificationHubNamespaceStateRefreshFunc(ctx context.Context, client notificationhubs.NamespacesClient, resourceGroupName string, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		res, err := client.Get(ctx, resourceGroupName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(res.Response) {
+				return nil, "", fmt.Errorf("Error retrieving Notification Hub Namespace %q (Resource Group %q): %s", name, resourceGroupName, err)
+			}
+		}
+
+		// try triggering a deleting again
+		client.Delete(ctx, resourceGroupName, name)
```

Comment:
just want to call this out as this looks odd on first glance - the Namespace delete method doesn't actually delete it some of the time, but repeatedly hitting it seems to work so 🙃